### PR TITLE
Only pre-ramp ramping releases

### PR DESCRIPTION
**Pull request by dokipen**
Only pre-ramp ramping releases

### CODE REVIEW

**Comment by dokipen** [comment]
:zap:

File: pkg/controller/releasemanager/syncer_test.go [hunk 1]
```diff
@@ -85,6 +89,7 @@ func createTestIncarnation(tag string, currentState State, currentPercent int, o
 				},
 			},
 		},
+		isRamping: isRamping,
```
Comment (silverlyra) [inline pkg/controller/releasemanager/syncer_test.go:92]:
```suggestion
		isRamping: currentState == releasing,
```

File: pkg/controller/releasemanager/syncer_test.go [hunk 2]
```diff
@@ -34,6 +34,10 @@ func (s testScale) Apply(i *Incarnation, currentPercent int) {
 }
 
 func createTestIncarnation(tag string, currentState State, currentPercent int, options ...testIncarnationOption) *Incarnation {
+	var isRamping bool
+	if currentState == releasing {
+		isRamping = true
+	}
```
Comment (silverlyra) [inline pkg/controller/releasemanager/syncer_test.go:40]:
```suggestion
```

File: pkg/controller/releasemanager/incarnation.go [hunk 3]
```diff
@@ -811,6 +812,17 @@ func newIncarnationCollection(controller Controller, revisionList *picchuv1alpha
 		add(r.Tag, nil)
 	}
 
+	// mark ramping incarnation
+	sorted := ic.releasable()
+	for i := range sorted {
+		inc := sorted[i]
+		state := State(inc.status.State.Current)
+		if state != canaried && state != canarying {
+			inc.isRamping = true
```
Comment (silverlyra) [inline pkg/controller/releasemanager/incarnation.go:821]: add a log?

**Pull request merged**